core/frontend/src/ViewState.ts

```diff
@@ -1040,7 +1040,7 @@ export abstract class ViewState extends ElementState {
    */
   public attachToViewport(_vp: Viewport): void {

```

Comment:
Would there be a reason (maybe just debugging?) to save the vp on the ViewState? I realize that that causes a cycle, so perhaps it's a bad idea for that reason.

Reply:
In our old C++ code a view stored a pointer to a DgnViewport and contained all kinds of logic that did different things based on whether or not that pointer was null. I wanted to avoid that here - the Viewport owns and manipulates the ViewState, never the other way around. I can add a comment explaining the condition inside `isAttachedToViewport`.

Reply:
I remember the C++ world and the who's-in-charge nightmare, so avoiding that is a good idea. But, then why pass it as an argument? Just inviting someone to save it.

Reply:
Sheet views and section drawing views will need access to it when they attach/detach. They won't need to store it.

Reply:
^ This may end up not being true now with the changes I've made. I'll find out when I hook up the attach/detach methods for those subclasses.

Reply:
Removed the Viewport arg.